src/aof.c

```diff
@@ -1010,16 +1010,22 @@ int startAppendOnly(void) {
  * the first call is short, there is a end-of-space condition, so the next
  * is likely to fail. However apparently in modern systems this is no longer
  * true, and in general it looks just more resilient to retry the write. If
- * there is an actual error condition we'll get it at the next try. */
-ssize_t aofWrite(int fd, const char *buf, size_t len) {
-    ssize_t nwritten = 0, totwritten = 0;
+ * there is an actual error condition we'll get it at the next try.
+ * We also check for aof-max-size limit here returning "no space" on exceed. */
+ssize_t aofWrite(int fd, const char *buf, size_t len, off_t aof_current_size, unsigned long long aof_max_size) {
+    ssize_t nwritten = 0, totwritten = 0, nonewritten = -1;
+
+    if (aof_max_size && (unsigned long long)aof_current_size >= aof_max_size) {
+        errno = ENOSPC;
```

Comment:
The error message:
`Error writing to the AOF file: <ENOSPC>`
is misleading if the issue is not actual disk space but hitting the configured max file size. Consider logging a more explicit message, and using `EFBIG` instead of `ENOSPC` for clarity.

Reply:
I thought about smth like that, but haven't found a better error code than ENOSPC. the one you offered is (https://man7.org/linux/man-pages/man2/open.2.html):
`       EOVERFLOW
              pathname refers to a regular file that is too large to be
              opened.  The usual scenario here is that an application
              compiled on a 32-bit platform without
              -D_FILE_OFFSET_BITS=64 tried to open a file whose size
              exceeds (1<<31)-1 bytes; see also O_LARGEFILE above.  This
              is the error specified by POSIX.1; before Linux 2.6.24,
              Linux gave the error EFBIG for this case.`

seems that it's more misleading (imho) - we're not opening a file, we're writing to it, and we're exactly out of space (though the limit is not a real disk limit).

we can try to define our own errno type, a new one, to differ it from the real disk limit and form a more descriptive message based on that upper in code stack - but shouldn't that be an overkill for such an issue?

Reply:
POSIX error codes aren't tied to specific system/library calls and can be applied broadly. In the example above, it was in the context of `open`. Applications can use their own interpretations, this is an example:
https://github.com/valkey-io/valkey/blob/unstable/src/module.c#L5427

I'm fine with using either error codes, but there should be a custom error message to avoid misleading users into thinking it's a real disk space issue if ENOSPC is used.